src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/service/PrisonerProfileService.kt

```diff
@@ -28,43 +29,25 @@ class PrisonerProfileService(
     val LOG: Logger = LoggerFactory.getLogger(this::class.java)
   }
 
-/*
-  TODO - keep this till we do the performance tests
-  fun getPrisonerProfile(offenderNo: String): PrisonerProfileDto? {
-    val prisoner = prisonerOffenderSearchClient
-      .getPrisonerById(offenderNo) ?: throw InvalidPrisonerProfileException("Unable to retrieve offender details from Prison Offender Search API")
-    val inmateDetail = prisonApiClient
-      .getInmateDetails(offenderNo) ?: throw InvalidPrisonerProfileException("Unable to retrieve inmate details from Prison API")
-    val visitBalances = prisonApiClient
-      .getVisitBalances(offenderNo) ?: throw InvalidPrisonerProfileException("Unable to retrieve visit balances from Prison API")
-
-    val prisonerBookingSummaryList =
-      prisonApiClient.getBookings(offenderNo, prisoner.prisonId)?.content
-    val prisonerBookingSummary = prisonerBookingSummaryList?.firstOrNull()
-
-    return PrisonerProfileDto(prisoner, inmateDetail, prisonerBookingSummary, visitBalances)
-  }
-*/
-
   fun getPrisonerProfile(prisonId: String, prisonerId: String): PrisonerProfileDto? {
-    val prisoner = prisonerOffenderSearchClient.getPrisonerById(prisonerId)
-    val inmateDetail = prisonApiClient.getInmateDetails(prisonerId)
-    val visitBalances = prisonApiClient.getVisitBalances(prisonerId)
-    val prisonerBookingSummaryList = prisonApiClient.getBookings(prisonId, prisonerId)
+    val prisonerPromise = prisonerOffenderSearchClient.getPrisonerById(prisonerId)
+    val inmateDetailPromise = prisonApiClient.getInmateDetails(prisonerId)
+    val visitBalancesPromise = prisonApiClient.getVisitBalances(prisonerId)
+    val prisonerBookingSummaryPromise = prisonApiClient.getBookings(prisonId, prisonerId)
 
     val prisonerProfileTuple =
-      Mono.zip(prisoner, inmateDetail, visitBalances, prisonerBookingSummaryList).block(apiTimeout)
+      Mono.zip(prisonerPromise, inmateDetailPromise, visitBalancesPromise, prisonerBookingSummaryPromise).block(apiTimeout)
     val prisonerProfile = getPrisonerProfileDto(prisonerProfileTuple)
     validatePrisonersPrisonId(prisonerProfile, prisonId)
     return prisonerProfile
```

Comment:
Wonder if it would be better to do 
`Mono.zip(prisonerPromise,inmateDetailPromise).zipWhen(`
your call